src/persist-client/src/impl/machine.rs

```diff
@@ -758,12 +775,14 @@ mod tests {
 
             let state = Arc::new(Mutex::new(DatadrivenState::default()));
             let cpu_heavy_runtime = Arc::new(CpuHeavyRuntime::new());
+            let now = SYSTEM_TIME.clone();
 
             f.run_async(move |tc| {
                 let shard_id = shard_id.clone();
                 let client = client.clone();
                 let state = Arc::clone(&state);
                 let cpu_heavy_runtime = Arc::clone(&cpu_heavy_runtime);
+                let now = now.clone();
```

Comment:
nit: I've been thinking we should stuff all this sort of thing in `state` so we only have one clone line up here (looks like I missed this in the CpuHeavyRuntime PR). but not tied to that if you're attached to doing it this way

Reply:
not attached to doing it this way, but opted to not tackle this right now. don't want to delay this one getting out the door -- writer leases will be very helpful for debugging live issues